### PR TITLE
fix: get rid of red error marker in get_components_from_service

### PR DESCRIPTION
**Pull request by SteBaum**
#### Which issue(s) this PR fixes

Fixes None

#### Additional comments

- Get rid of red error marker in get_components_from_service in Collections class.
- Get rid of red error marking in Dag class.

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Review by PaulFarault — CHANGES_REQUESTED** [review]
Agree with the `Dag` fix.

For the `Collections.get_components_from_service`, we may just want to remove the `Operation.is_service_operation` method entirely and replace it with what you suggested (checking the `Operation.component_name` value).

**Comment by SteBaum** [comment]
> Agree with the `Dag` fix.
> 
> For the `Collections.get_components_from_service`, we may just want to remove the `Operation.is_service_operation` method entirely and replace it with what you suggested (checking the `Operation.component_name` value).

Yes both are mutually exclusive so we do not need both. I removed `Operation.is_service_operation` with last force push.

**Review by PaulFarault — APPROVED** [review]

File: tdp/core/dag.py [hunk 1]
```diff
@@ -274,28 +275,29 @@ def warning(collection_name: str, message: str) -> None:
...
-            if (
-                dependency.endswith("_start")
-                and dependency_service != operation.service_name
-            ):
-                c_warning(
-                    f"Operation '{operation_name}' is in service '{operation.service_name}', depends on "
-                    f"'{dependency}' which is a start action in service '{dependency_service}' and should "
-                    f"only depends on start action within its own service"
-                )
+        if operation.collection_name is not None:
```
Comment (rpignolet) [inline tdp/core/dag.py:278]: `collection_name` is optional for an operation ?

Comment (SteBaum) [reply tdp/core/dag.py:278]: At the moment yes. In the class `Operation`:

```
def __init__(
        self,
        name: str,
        collection_name: Optional[str] = None,
```

Comment (PaulFarault) [reply tdp/core/dag.py:278]: Maybe you could invert the condition (using `continue`) to improve code and diff.

Comment (rpignolet) [reply tdp/core/dag.py:278]: I understand you want to fix red marker but these marker remind us that the code need refactor. With this fix we will not remember to remove this `if` when `collection_name` will be mandatory.

**Comment by PaulFarault** [comment]
I suggest we close this PR that only fixes the symptoms and not the main issues.

I opened #625, we'll need to discuss if we should make the `collection_name` mandatory for the `Operation`.

**Pull request merged**